packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -142,109 +94,493 @@ describe("Decentraland", () => {
+            toAddress,
+            SMALL_AMOUNT
+          );
+          await transferEthersAction.waitForNonceToUpdate();
+          await confirmEtherBalances(provider, [toAddress], [SMALL_AMOUNT]);
+
+          const transferManaAction = gnosisSafe.transferERC20(
+            MANA_ADDRESS,
+            toAddress,
+            manaAmountForShopping
+          );
+          await transferManaAction.waitForNonceToUpdate();
+          await confirmBalances(mana, [toAddress], [manaAmountForShopping]);
+
+          mana.setWallet(ephemeralWallet);
+          const approvalAction = mana.approve(
+            MARKETPLACE_ADDRESS,
+            manaAmountForShopping,
+            gasParams
+          );
+          await approvalAction.waitForNonceToUpdate();
+
+          const allowance = await mana.allowance(
+            ephemeralAddress,
+            MARKETPLACE_ADDRESS
+          );
+
+          expect(`${allowance}`).to.equal(`${manaAmountForShopping}`);
+        });
+
+        it("should buy an estate", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = estateForSale;
+
+          checkAsset(estate, mana, estateForSale, ephemeralAddress);
+
+          await confirmBalances(estate, [ephemeralAddress], [0]);
+
+          const fingerprint = await estate.getFingerprint(`${assetId}`);
+
+          marketplace.setWallet(ephemeralWallet);
+          const executeOrderAction = marketplace.safeExecuteOrder(
+            nftAddress,
+            `${assetId}`,
+            `${priceInWei}`,
+            `${fingerprint}`,
+            gasParams
+          );
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await confirmBalances(estate, [ephemeralAddress], [1]);
+        });
+
+        it("should buy a parcel of land", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = landForSale;
+
+          checkAsset(land, mana, landForSale, ephemeralAddress);
+
+          await confirmBalances(land, [ephemeralAddress], [0]);
+
+          // LANDRegistry contract doesn't implement getFingerprint function
+          const fingerprint = "0x";
+          marketplace.setWallet(ephemeralWallet);
+          const executeOrderAction = marketplace.safeExecuteOrder(
+            nftAddress,
+            `${assetId}`,
+            `${priceInWei}`,
+            `${fingerprint}`,
+            gasParams
+          );
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await confirmBalances(land, [ephemeralAddress], [1]);
+        });
+      });
 
-        await executeOrderAction.waitForNonceToUpdate();
+      // WIP: Not working because of gas issue on Marketplace.safeExecuteOrder() call
+      describe.skip("Using a Gnosis Safe wallet", () => {
```

Comment:
Should we say that we're using the funds directly from the Safe wallet to make the purchase to be more specific?

Reply:
Done.

Reply:
Thanks! Resolving